sdk/inc/azure/core/az_mqtt5_connection.h

```diff
@@ -64,7 +64,7 @@ enum az_event_type_mqtt5_connection
  * @brief Callback for MQTT 5 connection events.
  *
  */
-typedef az_result (*az_mqtt5_connection_callback)(az_mqtt5_connection* client, az_event event);
+typedef az_result (*az_mqtt5_connection_callback)(az_mqtt5_connection* client, az_event event, const void* event_callback_context);
```

Comment:
I'm not convinced on why this would be `const`. If the application wants to change the data, it should be able to do so - it's not the SDK's data to begin with.